Automattic/newspack-popups
# feat: lookup unknown email addresses in ESP to determine status

### All Submissions:






### Changes proposed in this Pull Request:



Intercept an email address provided by the user, and if that email address is not already known to be associated with a newsletter subscription for the current reader client ID, look up the subscription status in the active ESP in Newspack Newsletters. If the email address is a known contact, log a subscription event for segmentation purposes.

This could happen if the user has subscribed to a newsletter via offsite means, or via the site prior to the Reader Activation flow.

Note: may require different hooks per https://github.com/Automattic/newspack-plugin/issues/1791.

Awaiting resolution of https://github.com/Automattic/newspack-newsletters/pull/891 in order to support all ESPs (currently only ActiveCampaign supports the `get_contact_lists` method).

Closes #893.

### How to test the changes in this Pull Request:

1. Connect an ActiveCampaign account. Set ActiveCampaign as the active ESP in **Engagement > Newsletters** and enable at least one list.
2. In the ActiveCampaign dashboard, manually add an email address as a contact to one or more lists that are enabled on the site.
3. In a new incognito session, sign up with that same email address via the Newspack Registration block without signing up for any newsletters in the block.
4. Navigate to another page and confirm that a.) you're segmented as a subscriber (see prompts targeted to subscribers only) and b.) a `subscription` event is logged to the `wp_newspack_campaigns_reader_events` table with the relevant info: `context` should be the email address, and `value` should contain the ESP slug (`active_campaign`) and an array of list IDs the email address is subscribed to.

### Other information:
